### ISSUE TITLE
Refactor polling to fix issue with requests not stopping when auth blocked

### ISSUE THREAD

**Issue by jacobwindsor** [ebi-ait/dcp-ingest-central#99]
We were blocked by Elixir because of too many requests to the JWK endpoint. This is due to the polling done in submissions and project views which doesn't stop when the token expires. Some refactoring to use DataSources would be good here

**Comment by aaclan-ebi** [comment]
@jacobwindsor just curious, how are you gonna fix this issue?

**Comment by jacobwindsor** [comment]
@aaclan-ebi I've created SimpleDataSource and PaginatedDataSource. Refactoring all components to use these so polling is handled in one place. Then SimpleDataSource will check for auth error status and stop polling if so. Maybe with three retries?

**Comment by prabh-t** [comment]
this looks like a good plan. would like to see the PR for this. are we leaving the websockets for another ticket?

**Comment by jacobwindsor** [comment]
@prabh-t https://github.com/ebi-ait/ingest-ui/pull/75

Yes, I think we should leave websockets for another time

**Comment by aaclan-ebi** [comment]
The testing done by @mshadbolt is noted in the PR https://github.com/ebi-ait/ingest-ui/pull/75.
Closing this ticket now. Thanks @jacobwindsor and @mshadbolt !
